kds/kds.go

```diff
@@ -227,32 +254,63 @@ func ComposeTCBParts(parts TCBParts) (TCBVersion, error) {
 		check127("Spl4", parts.Spl4),
 		check127("TeeSpl", parts.TeeSpl),
 		check127("BlSpl", parts.BlSpl),
+		check127("FmcSpl", parts.FmcSpl),
 	); err != nil {
-		return TCBVersion(0), err
+		return tcbVersion, err
 	}
-	return TCBVersion(
-		(uint64(parts.UcodeSpl) << 56) |
+
+	switch parts.Version {
+	case TCBStructVersion0:
+		tcbVersion.Version = TCBStructVersion0
+		tcbVersion.TCB = (uint64(parts.UcodeSpl) << 56) |
 			(uint64(parts.SnpSpl) << 48) |
 			(uint64(parts.Spl7) << 40) |
 			(uint64(parts.Spl6) << 32) |
 			(uint64(parts.Spl5) << 24) |
 			(uint64(parts.Spl4) << 16) |
 			(uint64(parts.TeeSpl) << 8) |
-			(uint64(parts.BlSpl) << 0)), nil
+			(uint64(parts.BlSpl) << 0)
+	case TCBStructVersion1:
+		tcbVersion.Version = TCBStructVersion1
+		tcbVersion.TCB = (uint64(parts.UcodeSpl) << 56) |
+			(uint64(parts.SnpSpl) << 24) |
+			(uint64(parts.TeeSpl) << 16) |
+			(uint64(parts.BlSpl) << 8) |
+			(uint64(parts.FmcSpl) << 0)
+	default:
+		return tcbVersion, fmt.Errorf("unsupported TCB struct version: %v", parts.Version)
+	}
```

Comment:
I don't anticipate the need for an error here. It should be safe to default to one structure if it isn't explicitly the other version. @deeglaze thoughts on this?